Set/ViewController.swift

```diff
@@ -8,12 +8,451 @@
+            addGesturesRecognizers()
+            
+            // Hide gameOver label
+            gameOverLabel.isHidden = true
+            
+            // make the scoreLabels' borders transparent but with some width.
+            initiateScoreLabelBorders()
+            
+            // make sure all the setups above only happen once.
+            isJustInitiated = false
+ 
+        } else { // will happen every time the device will change orientation:
+            placeOpenCardsViewsOnGrid()
+        }
+    }
+    
+    /* -------------------
+     Functions for Buttons
+     ---------------------*/
+    
+    // Resets the relevant propeties for a new game:
+    @IBAction func newGamePressed(_ sender: UIButton) {
+        game = SetGame()
+        
+        // reset scores:
+        playerOneScore = 0
+        playerTwoScore = 0
+        matchesCounter = 0
+        
+        // hide gameOverLabel and display the openCardsCanvas:
+        openCardsCanvas.isHidden = false
+        gameOverLabel.isHidden = true
+        
+        initiateScoreLabelBorders()
+        
+        updateViewFromModel()
+    }
+    
+    @IBAction func dealButtonPressed(_ sender: UIButton) {
+        preformThreeCardsDealing()
+    }
 
-}
+    // Pressing the cheatButton on a player's turn automatically finds a match and ends the turn
+    // (no scores rewards or penalties).
+    @IBAction func cheatButtonPressed(_ sender: UIButton) {
+        guard !isAMatchMarked else {return} // if a match is marked, the game is paused, and we can ignore the pressing.
+        
+        game.resetCardSelection()
+        if let match = game.findMatchInOpenCards() {
+            for i in 0..<match.count {
+                game.chooseCard(chosenCard: match[i])
+            }
+        } else if !game.deck.isEmpty {
+            print("There isn't a 'set' in the currently open cards, so the game is opening more cards (if available) and finds a set")
+            preformThreeCardsDealing()
+            cheatButtonPressed(sender)
+        }
+        stopTimer()
+        updateViewFromModel()
+    }
+    
+    @IBAction func playerOneButtonPressed(_ sender: Any) {
+        playerButtonPressed(playerIdentifier: 1)
+        updateViewFromModel()
+    }
+    
+    @IBAction func playerTwoButtonPressed(_ sender: Any) {
+        playerButtonPressed(playerIdentifier: 2)
+        updateViewFromModel()
+    }
+    
+    /* ---------------
+     Gesture Handlers
+     ---------------- */
+    
+    // This function is called every time a user taps on a setCardView. It updates the model
+    // that a card was selected and it updates the view accordingly.
+    @objc private func cardTapHandler(recognizer: UITapGestureRecognizer) {
+        
+        if !isPlayerOneTurn && !isPlayerTwoTurn { // if this is already one of the players' turns, ignore
+            print("Tried to touch a card witout taking the turn (by pressing player 1 or player 2)")
```

Comment:
the same goes here, please use "Alert" for this message

Reply:
👍🏻👍🏻